### ISSUE TITLE
SQL path syntax and variable scope confusion

### ISSUE THREAD

**Issue by MarttiR** [orientechnologies/orientdb#9550]
### OrientDB Version: 3.1.8 (docker)

Variables with path syntax (like `$foo.bar`) get overridden by plain variables that have the same name as the path (e.g. `$bar`).

## Steps to reproduce

```sql
CREATE CLASS TestClass;
CREATE CLASS TestClassV EXTENDS V;

LET $foo = { "test": [ ["bar", "baz"], ["quux"] ] };
LET $test = ["a", "b", "c"];

CREATE VERTEX TestClassV SET
  a = $foo.test,
  b = $test;

INSERT INTO TestClass SET
  a = $foo.test,
  b = $test;

INSERT INTO TestClass CONTENT {
  "a": $foo.test,
  "b": $test
};

SELECT FROM TestClass WHERE a = $foo.test;
SELECT FROM TestClass WHERE b = $foo.test;

RETURN $foo.test;
```

## Expected behavior  
Expected the created records to have the fields `a` and `b` as they would logically follow from the variables used:
```
{
    "a": [
        [
          "bar",
          "baz"
        ],
        [
          "quux"
        ]
    ],
    "b": [
        "a",
        "b",
        "c"
    ]
}
```

Also expected the `SELECT .. WHERE` statements to use `$foo.test` instead of `$test` when filtering.

Also expected the `RETURN` to return the value in `$foo.test` and not the value in `$test`.

## Actual behavior  
All the created records contain the value of `$test` in both fields:
```
{
    "a": [
        "a",
        "b",
        "c"
    ],
    "b": [
        "a",
        "b",
        "c"
    ]
}
```

Also, the `SELECT` statements return the results that match the contents of `$test`, which is not correct.

Also, the `RETURN` statement returns the value of `$test` and not the value of `$foo.test`.

**Comment by MarttiR** [comment]
While I ran into this issue with variables containing arrays, the same thing happens with variables of any type – so it is definitely a scope issue. 

Minimal reproduction:
```sql
LET $foo = { "test": "is it working?" };
LET $test = "this is a bug";
RETURN $foo.test;
```

**Comment by MarttiR** [comment]
This can also bite from inside a `FOREACH`:

```sql
LET $foo = { "test": "correct value" };
LET $bar = ["bad value"];
LET $out = "";
FOREACH ($test in $bar) {
  LET $out = $out + $test;
}
RETURN $foo.test; /* returns "bad value" instead of "correct value" */
```

**Comment by luigidellaquila** [comment]
Hi @MarttiR 

I managed to reproduce and fix the problem, I'm pushing a fix now.
The fix will be released with v 3.1.12 (and with next 3.2.0 of course)

Thanks

Luigi
